exporters/trace/zipkin/model_test.go

```diff
@@ -780,3 +773,88 @@ func Test_toZipkinTags(t *testing.T) {
 		})
 	}
 }
+
+func TestRemoteEndpointTransformation(t *testing.T) {
+	tests := []struct {
+		name string
+		data *export.SpanSnapshot
+		want *zkmodel.Endpoint
+	}{
+		{
+			name: "nil-not-applicable",
+			data: &export.SpanSnapshot{
+				SpanKind:   trace.SpanKindClient,
+				Attributes: []attribute.KeyValue{},
+			},
+			want: nil,
+		},
+		{
+			name: "nil-not-found",
+			data: &export.SpanSnapshot{
+				SpanKind: trace.SpanKindConsumer,
+				Attributes: []attribute.KeyValue{
+					attribute.String("attr", "test"),
+				},
+			},
+			want: nil,
+		},
+		{
+			name: "peer-service-rank",
+			data: &export.SpanSnapshot{
+				SpanKind: trace.SpanKindProducer,
+				Attributes: []attribute.KeyValue{
+					semconv.PeerServiceKey.String("peer-service-test"),
+					semconv.NetPeerNameKey.String("peer-name-test"),
+					semconv.HTTPHostKey.String("http-host-test"),
+				},
+			},
+			want: &zkmodel.Endpoint{
+				ServiceName: "peer-service-test",
+			},
+		},
+		{
+			name: "net-peer-invalid-ip",
+			data: &export.SpanSnapshot{
+				SpanKind: trace.SpanKindProducer,
+				Attributes: []attribute.KeyValue{
+					semconv.NetPeerIPKey.String("INVALID"),
+				},
+			},
+			want: nil,
+		},
+		{
+			name: "net-peer-ipv6-no-port",
+			data: &export.SpanSnapshot{
+				SpanKind: trace.SpanKindProducer,
+				Attributes: []attribute.KeyValue{
+					semconv.NetPeerIPKey.String("0:0:1:5ee:bad:c0de:0:0"),
+				},
+			},
+			want: &zkmodel.Endpoint{
+				IPv6: net.ParseIP("0:0:1:5ee:bad:c0de:0:0"),
+			},
+		},
+		{
+			name: "net-peer-ipv4-port",
+			data: &export.SpanSnapshot{
+				SpanKind: trace.SpanKindProducer,
+				Attributes: []attribute.KeyValue{
+					semconv.NetPeerIPKey.String("1.2.3.4"),
+					semconv.NetPeerPortKey.Int(9876),
+				},
+			},
+			want: &zkmodel.Endpoint{
+				IPv4: net.ParseIP("1.2.3.4"),
+				Port: 9876,
+			},
+		},
+	}
```

Comment:
Can we add tests for `peer.hostname`, `peer.address`, `http.host`, and `db.name`?